mage/query-modules/python/gnn-link-prediction.md

```diff
@@ -0,0 +1,375 @@
+| `checkpoint_freq` | int | `5` | Select the number of epochs on which the model will be saved. The model is persisted on disc. |
+| `aggregator` | str | `mean` | Aggregator used in GraphSAGE model. Supported values are lstm, pool, mean and gcn. |
+| `metrics` | mgp.List[str] | `[loss, accuracy, auc_score, precision, recall, f1, true_positives, true_negatives, false_positives, false_negatives]` | Metrics used to evaluate the model in training on the validation set. Additionally, epoch information will always be displayed. |
+| `predictor_type` | str | `dot` | Type of the predictor. A predictor is used for combining node scores to edge scores. Supported values are `dot` and `mlp`. |
+| `attn_num_heads` | List[int] | `[4, 1]` | `GAT` can support the usage of more than one head in each layer except the last one. The size of the list must be the same as the number of layers specified by the `hidden_features_size` parameter. |
+| `tr_acc_patience` | int | `8` | Training patience, specifies how many epochs will accuracy drop on the validation set be tolerated before the training will be stopped. |
+| `context_save_dir` | str | `None` | Path where the model and predictor will be saved every checkpoint_freq epochs. |
+| `target_relation` | str | `None` | Unique edge type that is used for training. Users can provide only `edge_type` or `tuple of the source node, edge type, dest_node` if the same `edge_type` is used with more source-destination node combinations. |
+| `num_neg_per_pos_edge` | int | `1` | Number of negative edges that will be sampled per one positive edge in the mini-batch training. |
+| `batch_size` | int | `256` | Batch size used in both training and validation procedure. It specifies the number of indices in each batch. |
+| `sampling_workers` | int | `5` | Number of workers that will cooperate in the sampling procedure in the training and validation. |
+| `last_activation_function` | str | `sigmoid` | Activation function that is applied after the last layer in the model and before the `predictor_type`. Currently, only `sigmoid` is supported. | 
+| `add_reverse_edges` | bool | `False` | Whether the module should add reverse edges for each in the obtained graph. If the source and destination node are of the same type, edges of the same edge type will be created. If the source and destination nodes are different, then the prefix `rev_` will be added to the previous edge type. Reverse edges will be excluded as message passing edges for corresponding supervision edges. |
+
+#### **Output**:
+- `status: bool` -> True if all parameters were successfully updated, False otherwise.
+- `message: str` -> "OK" if all parameters were successfully updated, *Error message* otherwise. 
+
+__You__ can send only the parameters you want to change from the default ones, so e.g the method call could look like this:
```

Comment:
Only those parameters that need changing from their default values are sent when calling the procedure: